[BUG] Crash when retrieving docker apps with datasource_apps

### Is there an existing issue for this?

- [x] I have searched the existing issues

### What version of the Terraform provider are you using?

1.9.0

### What version of the Terraform CLI are you using?

1.13.4

### What CF API version are you using?

3.202.0

### What type of issue are you facing

bug report

### Describe the bug

I get the following stack trace when retrieving apps with `data.cloudfoundry_apps` and at least one app is based on a docker image and was pushed with `--docker-username`. If all apps are buildpack-based, or if the docker app was pushed anonymously, then the datasource can complete without crashing.

```
Stack trace from the terraform-provider-cloudfoundry_v1.9.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0xe0 pc=0x10506b050]

goroutine 846 [running]:
github.com/cloudfoundry/terraform-provider-cloudfoundry/cloudfoundry/provider.mapAppValuesToType({_, _}, _, _, _, _)
        github.com/cloudfoundry/terraform-provider-cloudfoundry/cloudfoundry/provider/types_app.go:405 +0x2d0
github.com/cloudfoundry/terraform-provider-cloudfoundry/cloudfoundry/provider.(*appsDataSource).Read(0x1400072c028, {0x1055f0b08, 0x14000e80b70}, {{{{0x1055f7b78, 0x14000f08d50}, {0x105490fe0, 0x14000f08660}}, {0x1055fde58, 0x14000e24040}}, {{{0x0, ...}, ...}, ...}, ...}, ...)
        github.com/cloudfoundry/terraform-provider-cloudfoundry/cloudfoundry/provider/datasource_apps.go:296 +0x710
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadDataSource(0x14000210c88, {0x1055f0b08, 0x14000e80b70}, 0x14000e8c200, 0x1400005f600)
        github.com/hashicorp/terraform-plugin-framework@v1.15.1/internal/fwserver/server_readdatasource.go:103 +0x5cc
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadDataSource(0x14000210c88, {0x1055f0b08?, 0x14000e80a80?}, 0x14000e80a20)
        github.com/hashicorp/terraform-plugin-framework@v1.15.1/internal/proto6server/server_readdatasource.go:55 +0x2b8
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadDataSource(0x14000269e00, {0x1055f0b08?, 0x14000e80270?}, 0x14000f16230)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/tf6server/server.go:733 +0x1b4
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadDataSource_Handler({0x1055bf660, 0x14000269e00}, {0x1055f0b08, 0x14000e80270}, 0x1400059c780, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:741 +0x1b8
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001a2a00, {0x1055f0b08, 0x14000e801e0}, 0x140001fc0c0, 0x1400036cc30, 0x105cf1e10, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0xbd0
google.golang.org/grpc.(*Server).handleStream(0x140001a2a00, {0x1055f1f70, 0x1400047e000}, 0x140001fc0c0)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0x858
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x74
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 39
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x120

Error: The terraform-provider-cloudfoundry_v1.9.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

I can use `datasource_apps` with authenticated-docker-based apps.

### Steps To Reproduce

1. Push a docker-based app to your space, such as:
    ```
    cf push test-tf-cli --docker-image hashicorp/terraform -c /bin/sh --no-route --health-check-type process --docker-username <redacted>
    ```
2. Run the following terraform config
3. View the crash backtrace
 
### Your Terraform Configuration

```hcl
terraform {
  required_version = "~> 1.13"
  required_providers {
    cloudfoundry = {
      source  = "cloudfoundry/cloudfoundry"
      version = "= 1.9.0"
    }
  }
}

provider "cloudfoundry" {}

data "cloudfoundry_org" "org" {
  name = "<redacted>"
}

data "cloudfoundry_space" "space" {
  org  = data.cloudfoundry_org.org.id
  name = "<redacted>"
}

data "cloudfoundry_apps" "apps" {
  org   = data.cloudfoundry_org.org.id
  space = data.cloudfoundry_space.space.id
}

check "validate_apps_count" {
  assert {
    condition     = length(data.cloudfoundry_apps.apps.apps) == 1
    error_message = "Only one app should be running"
  }
}
```

### Roles

- OrgManager
- SpaceDeveloper

### Add screenshots to help explain your problem

_No response_

### Additional context

_No response_

--- a/cloudfoundry/provider/datasource_app.go
+++ b/cloudfoundry/provider/datasource_app.go
@@ -306,7 +306,7 @@ func (d *appDataSource) Read(ctx context.Context, req datasource.ReadRequest, re
 		resp.Diagnostics.AddError("Error unmarshalling app", err.Error())
 		return
 	}
-	atResp, diags := mapAppDatasourceValuesToType(ctx, appManifest.Applications[0], app, nil, sshResp)
+	atResp, diags := mapAppDatasourceValuesToType(ctx, appManifest.Applications[0], app, sshResp)
 	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return

--- a/cloudfoundry/provider/datasource_apps.go
+++ b/cloudfoundry/provider/datasource_apps.go
@@ -293,7 +293,7 @@ func (d *appsDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 			return
 		}
 
-		atResp, diags := mapAppDatasourceValuesToType(ctx, appManifest.Applications[0], app, nil, sshResp)
+		atResp, diags := mapAppDatasourceValuesToType(ctx, appManifest.Applications[0], app, sshResp)
 		resp.Diagnostics.Append(diags...)
 		if resp.Diagnostics.HasError() {
 			return

--- a/cloudfoundry/provider/types_app.go
+++ b/cloudfoundry/provider/types_app.go
@@ -739,7 +739,7 @@ func setEnvForUpdate(ctx context.Context, existingEnvs basetypes.MapValue, plann
 	return finalEnvs, diagnostics
 }
 
-func mapAppDatasourceValuesToType(ctx context.Context, appManifest *cfv3operation.AppManifest, app *cfv3resource.App, reqPlanType *AppType, sshResp *cfv3resource.AppFeature) (DatasourceAppType, diag.Diagnostics) {
+func mapAppDatasourceValuesToType(ctx context.Context, appManifest *cfv3operation.AppManifest, app *cfv3resource.App, sshResp *cfv3resource.AppFeature) (DatasourceAppType, diag.Diagnostics) {
 	var diags, tempDiags diag.Diagnostics
 	var appType DatasourceAppType
 	appType.Name = types.StringValue(appManifest.Name)

